thor/params.go

```diff
@@ -77,6 +78,9 @@ func GetMaxBlockProposers(params interface {
 		return 0, err
 	}
 
+	if !mbp.IsUint64() {
```

Comment:
This might prevent the chain from progressing, not possible in real world but just for safety, how about return MaxUint64 here.